reconcile/utils/external_resources.py

```diff
@@ -66,14 +65,14 @@ def managed_external_resources(namespace_info: Mapping[str, Any]) -> bool:
 
 
 def get_inventory_count_combinations(
-    inventory: ExternalResourceSpecInventory,
+    inventory: MutableMapping,
```

Comment:
nit: why mutable is needed? It seems we do not expect any mutation here and `Mapping` might suffice for this case?

Reply:
You are right. I used the parent class of the two classes used with this method, but is not necessary to use Mutable in this case. Thanks for catching it.

Reply:
`Mapping` is too generic as we need to ensure key type match to use `(k.provision_provider, k.provisioner_name, k.provider) for k in inventory`, can we extract a common interface for it?

Reply:
I Switched it to a inventories static list.